ddtrace/internal/sampling.py

```diff
@@ -208,44 +211,90 @@ def apply_span_sampling_tags(self, span):
 
 def get_span_sampling_rules():
     # type: () -> List[SpanSamplingRule]
-    json_rules_raw = os.getenv("DD_SPAN_SAMPLING_RULES")
-    if json_rules_raw is None:
+    json_rules = get_span_sampling_json()
+    sampling_rules = []
+    for rule in json_rules:
+        if not isinstance(rule, dict):
+            raise TypeError("rule specified via DD_SPAN_SAMPLING_RULES is not a dictionary:%r" % rule)
+        # If sample_rate not specified default to 100%
+        sample_rate = float(rule.get("sample_rate", 1.0))
+        service = rule.get("service")
+        name = rule.get("name")
+        # If max_per_second not specified default to no limit
+        max_per_second = int(rule.get("max_per_second", -1))
+        if service is not None and not isinstance(service, string_types):
+            raise ValueError("The service value is not a string or None:%r" % service)
+        if name is not None and not isinstance(name, string_types):
+            raise ValueError("The name value is not a string or None:%r" % name)
+
+        if service is None and name is None:
+            raise ValueError("Neither service or name specified for single span sampling rule:%r" % rule)
+        if service:
+            _check_unsupported_pattern(service)
+        if name:
+            _check_unsupported_pattern(name)
+
+        try:
+            sampling_rule = SpanSamplingRule(
+                sample_rate=sample_rate, service=service, name=name, max_per_second=max_per_second
+            )
+        except Exception as e:
+            raise ValueError("Error creating single span sampling rule {}: {}".format(json.dumps(rule), e))
+        sampling_rules.append(sampling_rule)
+    return sampling_rules
+
+
+def get_span_sampling_json():
+    # type: () -> List[Dict[str, Any]]
+    env_json_rules = get_env_json()
+    file_json_rules = get_file_json()
+
+    if env_json_rules and file_json_rules:
+        log.warning(
+            (
+                "DD_SPAN_SAMPLING_RULES and DD_SPAN_SAMPLING_RULES_FILE detected. "
+                "Defaulting to DD_SPAN_SAMPLING_RULES value."
+            )
+        )
+        return env_json_rules
+    elif env_json_rules:
+        return env_json_rules
+    elif file_json_rules:
+        return file_json_rules
+    # No rules specified
+    else:
         return []
```

Comment:
This could be simplified to:
```suggestion
        return env_json_rules or file_json_rules or []
```

Reply:
😮